### PR TITLE
Remove deprecated PluginFactory classes #1584

### LINKED ISSUE #1584: Remove deprecated PluginFactory classes
**Is your feature request related to a problem? Please describe.**

The new plugin framework replaces the static PluginFactory classes. Delete these.

**Describe the solution you'd like**
Delete deprecated plugin classes and related testing code.

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Removes classes deprecated in Data Prepper 1.x.
 
### Issues Resolved

Resolves #1584.
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by asifsmohammed — APPROVED** [review]
LGTM

File: data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/TestSink.java [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-import com.amazon.dataprepper.model.record.Record;
-import com.amazon.dataprepper.model.sink.Sink;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.stream.Collectors;
-
-@DataPrepperPlugin(name = "test-sink", pluginType = Sink.class)
-public class TestSink implements Sink<Record<String>> {
```
Comment (cmanning09) [inline data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/TestSink.java:19]: Why are we deleting the TestSink test cases? The TestSink.class still exists and is not removed in this PR. 

https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/TestSink.java

Comment (cmanning09) [reply data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/TestSink.java:19]: The TestPrepper and TestSource test classes are also marked for removal but the implemented classes still exist.

Comment (cmanning09) [reply data-prepper-plugins/common/src/test/java/com/amazon/dataprepper/plugins/sink/TestSink.java:19]: Oh disregard, I have my paths mixed up

**Review by cmanning09 — APPROVED** [review]

**Pull request merged**